### DISCUSSION TITLE
Is it possible to bevel the "unknown" sharp edge generated by the thicken operation

### DISCUSSION THREAD

**Discussion by 1939938853** [jimy-byerley/pymadcad#93] [Q&A]
Hello

Is it possible to bevel the sharp edge generated by the thicken operation? I saw Blender can set roundness here in thicken operation. For Madcad, perhaps need to extract all boundary vertices first and then apply Madcad bevel operation?

**Comment by zrggci** [comment]
Have you found a solution? I am facing the same problem as you.I want to smooth the sharp edges, and it would be even better if the transition could be smooth like a rounded corner
![图片1(2)](https://github.com/jimy-byerley/pymadcad/assets/82247907/8bf1765c-ca5a-4dd8-95d7-e3945dd4ad6f)

**Reply by jimy-byerley** [reply]
You can use [`bevel()`](https://pymadcad.readthedocs.io/en/latest/reference/cut.html#madcad.cut.bevel)
But looking at your thickened mesh, it seems you missed the triangle connecting step after importing an STL file. take a look at `mergeclose` like in [the guide](https://pymadcad.readthedocs.io/en/latest/guide/io.html)

**Reply by zrggci** [reply]
Thank you for your answer. As you know, I am unable to obtain the boundary of the model，i upload the model, you can try
[thicken.zip](https://github.com/jimy-byerley/pymadcad/files/14119055/thicken.zip)
![41EA79C6-E18A-4a91-ACF1-4F251A3C57FB](https://github.com/jimy-byerley/pymadcad/assets/82247907/eba58eb5-107f-4298-a77b-059876ce979e)
I want to achieve this effect, it can transition smoothly
